File: aula_05/src/main/kotlin/com/branas/cleanarch/domain/driver/Driver.kt
package com.branas.cleanarch.domain.driver

import com.branas.cleanarch.application.repository.entities.DriverEntity
import com.branas.cleanarch.domain.Document
import com.branas.cleanarch.domain.Email
import java.time.LocalDateTime
import java.util.*

data class Driver(
    val id: UUID? = null,
    var name: String,
    var email: Email,
    var document: Document,
    var carPlate: CarPlate,
    var createdAt: LocalDateTime,
    val updatedAt: LocalDateTime? = null
) {
    fun toEntity() = DriverEntity(
        name = name,
        carPlate = carPlate.value,
        document = document.value,
        email = email.value,
        createdAt = createdAt,
        updatedAt = updatedAt,
    )

    companion object {
        fun create(name: String, email: String, carPlate: String, document: String, createdAt: LocalDateTime) = Driver(
            name = name,
            email = Email(email),
            carPlate = CarPlate(carPlate),
            document = Document(document),
            createdAt = createdAt
        )
    }
}

File: aula_06/frontend/src/domain/Driver.js
import CarPlate from "./CarPlate.js";
import Cpf from "./Cpf.js";
import Email from "./Email.js";
import Name from "./Name.js";

export default class Driver{

    constructor(driverId, name, email, document, carPlate) {  
      this.driverId = driverId;
      this.name = new Name(name);
      this.email = new Email(email);
      this.document = new Cpf(document);
      this.carPlate = new CarPlate(carPlate);
      }

      static create (builder) {
        return new Driver(builder.driverId, builder.name, builder.email, builder.document, builder.carPlate);
      }
}

export class DriverBuilder {
  constructor(name, email, document, carPlate) {  
    this.name = name;
    this.email = email;
    this.document = document;
    this.carPlate = carPlate;
    }
  
	build () {
		return Driver.create(this);
	}
}
